### PR TITLE
feat: sign changed files

### PR DESCRIPTION
**Pull request by eddycharly**
Signed-off-by: Charles-Edouard Brétéché <charles.edouard@nirmata.com>

### CODE REVIEW

File: .github/workflows/sign.yaml [hunk 1]
```diff
@@ -3,58 +3,52 @@ name: Test
...
 on:
   pull_request:
-    types: [ closed ]
+    branches:
+      - main
 
 jobs:
   merge_job:
-    # this job will only run if the PR has been merged
-    if: github.event.pull_request.merged == true
```
Comment (chipzoller) [inline .github/workflows/sign.yaml:11]: Why remove this? The signing should only be done when the PR has been approved and merged.

File: .github/workflows/sign.yaml [hunk 2]
```diff
@@ -3,58 +3,52 @@ name: Test
...
     runs-on: ubuntu-latest
     permissions:
       contents: write
       actions: read
       id-token: write # Needed for OIDC and keyless signing
     steps:
-    - run: |
-        echo PR #${{ github.event.number }} has been merged, beginning signing flow
```
Comment (chipzoller) [inline .github/workflows/sign.yaml:19]: Why remove this? The signing should only be done when the PR has been approved and merged.

File: .github/workflows/sign.yaml [hunk 3]
```diff
@@ -3,58 +3,52 @@ name: Test
...
       uses: actions/checkout@v3
       with:
+        ref: ${{ github.head_ref }}
         fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+    - name: Get changed files in the manifests folder
+      id: changed-files
+      uses: tj-actions/changed-files@v35
+      with:
+        files: manifests/*.{yaml,yml}
+    - name: List all changed files
```
Comment (chipzoller) [inline .github/workflows/sign.yaml:27]: What I was thinking was you don't want anyone to modify an unsigned PE, you always want them to create a new one. So we should only look for _new_ files in that case. And this is why in my action I was trying to clear the `/unsigned` dir after each successfully merged PR to empty out the dropbox.

**Review by chipzoller — COMMENTED** [review]
Commented

File: .github/workflows/sign.yaml [hunk 4]
```diff
@@ -3,58 +3,52 @@ name: Test
 # only trigger on pull request closed events
 on:
   pull_request:
-    types: [ closed ]
+    branches:
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply .github/workflows/sign.yaml:6]: Is this a strict requirement ? Signing in the pr branch will have side effects ?

File: .github/workflows/sign.yaml (hunk 1 above)
Comment (eddycharly) [reply .github/workflows/sign.yaml:11]: Related to the change above.

File: .github/workflows/sign.yaml (hunk 3 above)
Comment (eddycharly) [reply .github/workflows/sign.yaml:27]: How are you going to enforce this ?

File: .github/workflows/sign.yaml [hunk 5]
```diff
@@ -3,58 +3,52 @@ name: Test
...
+        done
     - name: Sign
       run: |
         curl -sLO https://github.com/sigstore/k8s-manifest-sigstore/releases/download/v0.4.3/kubectl-sigstore-linux-amd64
         chmod +x kubectl-sigstore-linux-amd64
         sudo mv kubectl-sigstore-linux-amd64 kubectl-sigstore
         ./kubectl-sigstore version
         ls -lah
-        for f in $(ls ./unsigned/)
+        for file in ${{ steps.changed-files.outputs.all_changed_files }}
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply .github/workflows/sign.yaml:39]: It's there:
```yaml
    - name: Get changed files in the manifests folder
      id: changed-files
      uses: tj-actions/changed-files@v35
      with:
        files: manifests/*.{yaml,yml}
```

File: .github/workflows/sign.yaml [hunk 6]
```diff
@@ -3,58 +3,52 @@ name: Test
...
         do
           echo "Signing $f"
-          COSIGN_EXPERIMENTAL=1 ./kubectl-sigstore sign --tarball=no -f unsigned/$f -o signed/$f
+          COSIGN_EXPERIMENTAL=1 ./kubectl-sigstore sign --tarball=no -f $file -o $file
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply .github/workflows/sign.yaml:42]: Can't they be signed in place ?

**Pull request closed without merging**